### PR TITLE
Don't crash the server if a connection filter throws synchronously.

### PR DESCRIPTION
**Pull request by cesarbs**
https://github.com/aspnet/KestrelHttpServer/issues/370

cc @halter73 @Tratcher

### CODE REVIEW

File: test/Microsoft.AspNet.Server.KestrelTests/ConnectionFilterTests.cs [hunk 1]
```diff
@@ -80,7 +81,36 @@ public class ConnectionFilterTests
                         "Hello World!");
                 }
             }
-       }
+        }
+
+        [ConditionalFact]
+        [FrameworkSkipCondition(RuntimeFrameworks.Mono, SkipReason = "Test hangs after execution on Mono.")]
```
Comment (halter73) [inline test/Microsoft.AspNet.Server.KestrelTests/ConnectionFilterTests.cs]: We no longer need this skip condition, right? Though I guess we should remove them all in a separate commit.

File: test/Microsoft.AspNet.Server.KestrelTests/ConnectionFilterTests.cs [hunk 2]
```diff
@@ -80,7 +81,36 @@ public class ConnectionFilterTests
...
+                    {
+                        await connection.SendEnd(
+                            "POST / HTTP/1.0",
+                            "",
+                            "Hello World?");
+                    }
+                    catch (IOException)
+                    {
+                        // Will throw because the exception in the connection filter will close the connection.
+                        Assert.True(true);
```
Comment (halter73) [inline test/Microsoft.AspNet.Server.KestrelTests/ConnectionFilterTests.cs]: Did this test fail before your change? If I remember correctly, the entire server would crash in the scenario previously, so I would expect an exception from SendEnd before this change.

Maybe before TestServer.Dispose was throwing?

Comment (cesarbs) [inline test/Microsoft.AspNet.Server.KestrelTests/ConnectionFilterTests.cs]: Before the change the xunit runner crashes along with the server :)

File: test/Microsoft.AspNet.Server.KestrelTests/ConnectionFilterTests.cs (hunk 1 above)
Comment (cesarbs) [inline test/Microsoft.AspNet.Server.KestrelTests/ConnectionFilterTests.cs]: Yeah, it looks like the problem is gone. I'd like to do a little more testing though.

File: test/Microsoft.AspNet.Server.KestrelTests/ConnectionFilterTests.cs (hunk 2 above)
Comment (halter73) [inline test/Microsoft.AspNet.Server.KestrelTests/ConnectionFilterTests.cs]: :laughing:

**Comment by halter73** [comment]
Oops. I thought I gave this a :ship: last night. :shipit:

**Pull request closed without merging**